### ISSUE TITLE
Clarity function to merge record structures

### ISSUE THREAD

**Issue by njordhov** [blockstack/stacks-blockchain#1733]
**Is your feature request related to a problem? Please describe.**

**Clarity record structures** (aka _tuples_) are currently cumbersome to merge, requiring constructing a new record by explicitly specifying all fields and their values, including extracting fields from the merged record(s). Issue #1728 links to several examples of such manually constructed merges in the context of updating persistent maps, such as:

https://github.com/psq/swapr/blob/878d3f3eaf3d651d8360bca13309c1268d2872d0/contracts/swapr.clar#L500-L519

The referenced example demonstrates the complexity of just updating a record with a new value for a single field (`fee-to-address)`:

```scheme
(let ((pair (unwrap! (map-get? pairs-data-map ((token-x token-x) (token-y token-y))) invalid-pair-err)))
  (map-set pairs-data-map ((token-x token-x) (token-y token-y))
    ((shares-total (get shares-total pair))
     (balance-x (get balance-x pair))
     (balance-y (get balance-y pair))
     (fee-balance-x (get fee-balance-y pair))
     (fee-balance-y (get fee-balance-y pair))
     (fee-to-address none))))
```

**Describe the solution you'd like**

Clarity should have a **function to merge two or more record structures**, returning a new record with the combined fields without mutating the original records. The records are combined left to right, with fields in the rightmost arguments taking precedence.

Here is how the example above can be simplified using `merge`:

```clojure
(let ((pair (unwrap! (map-get? pairs-data-map ((token-x token-x) (token-y token-y))) invalid-pair-err)))
  (map-set pairs-data-map ((token-x token-x) (token-y token-y))
    (merge pair { fee-to-address: none })))
```

**Additional context**

Other languages have functions to merge records (or similar types). The Clojure lisp dialect has [merge](https://clojuredocs.org/clojure.core/merge). Typed Racket lisp has [record-merge2](https://docs.racket-lang.org/rebellion/Records.html#%28def._%28%28lib._rebellion%2Fcollection%2Frecord..rkt%29._record-merge2%29%29). Ruby has [merge](https://docs.ruby-lang.org/en/2.0.0/Hash.html#method-i-merge). In Javascript, Immutable has [merge](https://immutable-js.github.io/immutable-js/docs/#/Map/merge); Lodash FP has [merge](https://github.com/lodash/lodash/wiki/FP-Guide).

**Comment by psq** [comment]
if I could give this proposal 2 thumbs up, I would.  It solves the use cases https://github.com/blockstack/stacks-blockchain/issues/1728 was proposing to address, and is more versatile.

It also has a more functional feel to it, and makes the merging explicit, which makes it more readable as well.

**Comment by kantai** [comment]
This is a good suggestion @njordhov -- `merge` should be implementable pretty directly, and I'd support and be happy to review a PR that implements this.

**Comment by lgalabru** [comment]
Addressed with #2117
